src/ArrayManager.hpp

```diff
@@ -50,6 +50,14 @@
 
 namespace chai {
 
+#if defined(CHAI_ENABLE_PICK_SET_INCR_DECR)
+#define DTOH 1
+#define HTOD 0
+
+#define __DTOH__(X) X == DTOH
```

Comment:
I think having an enum for direction is better than macros. You can even use an enum class.